src/components/Sidenav/Sidenav.tsx

```diff
@@ -0,0 +1,247 @@
+
+import { SidenavContext } from './Sidenav.context';
+
+import { useMenuAim, useWindowEventListener } from '../../hooks';
+
+type SidenavOwnerState = {
+  classes?: SidenavProps['classes'];
+  hover?: boolean;
+  open?: boolean;
+};
+
+const useUtilityClasses = (ownerState: SidenavOwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['root'],
+    container: ['container'],
+    rail: ['rail'],
+    drawer: ['drawer'],
+    overlay: ['overlay']
+  };
+
+  return composeClasses(slots, getSidenavUtilityClass, classes);
+};
+
+const SidenavRoot = styled('div', {
+  name: 'ESSidenav',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root
+})(() => ({
+  position: 'sticky',
+  top: '0',
+  height: '100%'
+}));
+
+const SidenavContainer = styled('div', {
+  name: 'ESSidenav',
+  slot: 'Container',
+  overridesResolver: (props, styles) => styles.container
+})(() => ({
+  display: 'flex',
+  height: '100%'
+}));
+
+const SidenavOverlay = styled('div', {
+  name: 'ESSidenav',
+  slot: 'Overlay',
+  overridesResolver: (props, styles) => styles.overlay
+})<{ ownerState: SidenavOwnerState }>(({ ownerState, theme }) => ({
+  opacity: '0',
+  userSelect: 'none',
+  top: '0',
+  position: 'absolute',
+  height: '100%',
+  width: '100vw',
+  backgroundColor: theme.palette.overlay[200],
+  transition: 'opacity 0.3s',
+
+  ...(ownerState.hover && {
+    opacity: '1'
+  }),
+
+  [theme.breakpoints.down('desktopXS')]: {
+    ...(ownerState.open && {
+      opacity: '1'
+    })
+  }
+}));
+
+const SidenavRail = styled('div', {
+  name: 'ESSidenav',
+  slot: 'Rail',
+  overridesResolver: (props, styles) => styles.rail
+})(({ theme }) => ({
+  display: 'flex',
+  height: '100%',
+  width: '57px',
+  zIndex: '99',
+
+  [theme.breakpoints.down('desktopXS')]: {
+    '& .ESSidebarToggle-button': {
+      display: 'none'
+    }
+  }
+}));
+
+const SidenavDrawer = styled('div', {
+  name: 'ESSidenav',
+  slot: 'Drawer',
+  overridesResolver: (props, styles) => styles.drawer
+})<{ ownerState: SidenavOwnerState }>(({ ownerState, theme }) => ({
+  position: 'absolute',
+  top: '0',
+  left: '-57px',
+  transform: 'translateX(-100%)',
+  zIndex: '98',
+  height: '100%',
+  transition: 'all 0.2s',
+
+  ...((ownerState.hover || ownerState.open) && {
+    left: '57px',
+    transform: 'translateX(0)'
+  }),
+
+  ...(ownerState.open && {
+    position: 'static'
+  }),
+
+  [theme.breakpoints.down('desktopXS')]: {
+    ...(ownerState.open && {
+      position: 'absolute'
+    })
+  }
+}));
+
+/**
+ * The Sidenav component is a fixed-position toggleable slide out box.
+ */
+export const Sidenav = (inProps: SidenavProps) => {
+  const { className, children, open, sx, disableEscapeKeyDown, onClose, ...props } = useThemeProps({
+    props: inProps,
+    name: 'ESSidenav'
+  });
+
+  const [hover, setHover] = useState(false);
+  const [itemIndex, setIndex] = useState<string | null>(null);
```

Comment:
Надо бы переименовать, так как сюда записывается не индекс, а айдишник, который строкойвой и там что угодно может быть. Как вариант:
```suggestion
  const [id, setId] = useState<string | null>(null);
```

Reply:
переименовал на _itemId_ , чтобы не было конфликта в имени с _id_ , который прилетает айтему через пропсы)